newdle/client/src/components/NewdlesParticipating.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import flask from 'flask-urls.macro';
+import {Container, Icon, Label, Placeholder} from 'semantic-ui-react';
+import {useHistory} from 'react-router';
+import {serializeDate, toMoment} from '../util/date';
+import client from '../client';
+import {usePageTitle} from '../util/hooks';
+import styles from './MyNewdles.module.scss';
+
+export default function NewdlesParticipating() {
+  const [participations, loading] = client.useBackend(() => client.getNewdlesParticipating(), []);
+  usePageTitle("Newdles I'm participating in");
+
+  let content;
+  if (loading || participations === null) {
```

Comment:
I am wondering whether `participations === null` should be another `else if` clause. Because e.g. if `loading === false` and for some reason `participations` will evaluate to `null` we will end up with placeholders which will not properly show the current state of the application. what about adding `else if (participations === null) return null`

Reply:
I agree, `participations === null` shouldn't show a loading state, this part was cloned from the previous implemented `MyNewdles.js`. However, I've been experimenting with both approaches and noticed the `loading` state isn't being enabled in this view because the backend reducer is expecting a `'submit'` action type and none is triggered for this fetch request. Just a `'success'` at the end of the call.

```
function backendReducer(state, action) {
  switch (action.type) {
    case 'submit':
      return {...state, submitting: true, error: '', result: null};
    case 'error':
      return {...state, submitting: false, error: action.error};
    case 'success':
      return {...state, submitting: false, result: action.result};
    default:
      return state;
  }
}
```

In practice, participations should always return empty, except when an error happens. So I'm not sure what is worth changing in this case, either:
- Refactor the loading state to always dispatch a 'submit' action before the API call and  'success'.
- Use the `participations === null` both as a loading and error state.